react/internal/http-switch-link.js

```diff
@@ -30,11 +29,13 @@ const hasMutationField = (queryTree) => {
 
 export const createHttpSwitchLink = (uri) => {
   const parsedUri = parse(uri, true)
+  // delete search in order to format to work as expected
+  parsedUri.search && delete parsedUri.search
```

Comment:
Can you explain the use case where this is important? That's not clear to me.

By the way, comments are most useful to explain things such as this: why the code is there, instead of just what it's doing (since the latter can be deduced by reading the code, but the former not always can).

Reply:
There is a bug (or a feature) in the url parser package that we are using that, if the field `search` is present in the object, the formatter won't use the `query` field for generating the URI. Since we want to force the formatter to use the `query` field, we need to drop the `search` field

Reply:
But then we'll just lose the information in this `search` field? What is it usually used for? Can we keep it in some other way?